package.json

```diff
@@ -58,7 +58,8 @@
   "lint-staged": {
     "*.{js,jsx,ts,tsx}": [
       "npm run format:provided",
-      "npm run lint:js"
+      "tsc --noEmit",
+      "eslint"
```

Comment:
I messed around with a code a bit and noticed the following:

1. There's a complaint in tsc if js file is staged. `--allowJs` option solves it.
2. Ideally we don't need to run tsc for Js at all. 

So I'd suggest updating config as follows:
```
"lint-staged": {
    "*.{js,jsx}": [
      "npm run format:provided",
      "eslint"
    ],
    "*.{ts,tsx}": [
      "npm run format:provided",
      "tsc --noEmit",
      "eslint"
    ],
....
```

It adds a bit of duplication but provides clear distinction between js and ts files processing.

Reply:
After a quick test, I think we can use for the second approach the following code:
```json
    "*.{js,jsx,ts,tsx}": [
      "npm run format:provided",
      "eslint"
    ],
    "*.{ts,tsx}": [
      "tsc --noEmit"
    ],
```
It's a bit faster than running eslint and prettier 2 times.

Reply:
Looks even better! :shipit: